contracts/AllowedRecipientsBuilder.sol

```diff
@@ -155,185 +175,159 @@ contract AllowedRecipientsBuilder {
 
-        assert(registry.getAllowedRecipients().length == _recipients.length);
+        assert(recipientRegistry.getAllowedRecipients().length == _recipients.length);
 
         for (uint256 i = 0; i < _recipients.length; i++) {
-            assert(registry.isRecipientAllowed(_recipients[i]));
+            assert(recipientRegistry.isRecipientAllowed(_recipients[i]));
+        }
+
+        tokenRegistry = factory.deployAllowedTokensRegistry(
+            admin, addRecipientToAllowedListRoleHolders, removeRecipientFromAllowedListRoleHolders
+        );
+
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            tokenRegistry.addToken(_tokens[i]);
         }
+        tokenRegistry.renounceRole(ADD_TOKEN_TO_ALLOWED_LIST_ROLE, address(this));
 
-        registry.setLimitParameters(_limit, _periodDurationMonths);
-        registry.renounceRole(SET_PARAMETERS_ROLE, address(this));
+        assert(tokenRegistry.getAllowedTokens().length == _tokens.length);
 
-        (uint256 registryLimit, uint256 registryPeriodDuration) = registry.getLimitParameters();
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            assert(tokenRegistry.isTokenAllowed(_tokens[i]));
+        }
+
+        recipientRegistry.setLimitParameters(_limit, _periodDurationMonths);
+        recipientRegistry.renounceRole(SET_PARAMETERS_ROLE, address(this));
+
+        (uint256 registryLimit, uint256 registryPeriodDuration) = recipientRegistry.getLimitParameters();
         assert(registryLimit == _limit);
         assert(registryPeriodDuration == _periodDurationMonths);
 
-        registry.updateSpentAmount(_spentAmount);
-        registry.renounceRole(UPDATE_SPENT_AMOUNT_ROLE, address(this));
+        recipientRegistry.updateSpentAmount(_spentAmount);
+        recipientRegistry.renounceRole(UPDATE_SPENT_AMOUNT_ROLE, address(this));
 
-        assert(registry.spendableBalance() == _limit - _spentAmount);
+        assert(recipientRegistry.spendableBalance() == _limit - _spentAmount);
 
-        assert(registry.hasRole(ADD_RECIPIENT_TO_ALLOWED_LIST_ROLE, admin));
-        assert(registry.hasRole(REMOVE_RECIPIENT_FROM_ALLOWED_LIST_ROLE, admin));
-        assert(registry.hasRole(SET_PARAMETERS_ROLE, admin));
-        assert(registry.hasRole(UPDATE_SPENT_AMOUNT_ROLE, admin));
-        assert(registry.hasRole(DEFAULT_ADMIN_ROLE, admin));
+        assert(recipientRegistry.hasRole(ADD_RECIPIENT_TO_ALLOWED_LIST_ROLE, admin));
+        assert(recipientRegistry.hasRole(REMOVE_RECIPIENT_FROM_ALLOWED_LIST_ROLE, admin));
+        assert(recipientRegistry.hasRole(SET_PARAMETERS_ROLE, admin));
+        assert(recipientRegistry.hasRole(UPDATE_SPENT_AMOUNT_ROLE, admin));
+        assert(recipientRegistry.hasRole(DEFAULT_ADMIN_ROLE, admin));
+        assert(tokenRegistry.hasRole(ADD_TOKEN_TO_ALLOWED_LIST_ROLE, admin));
+        assert(tokenRegistry.hasRole(REMOVE_TOKEN_FROM_ALLOWED_LIST_ROLE, admin));
```

Comment:
Seems like the check for DEFAULT_ADMIN_ROLE for the tokensRegistry is missing here and below